json/src/ser.rs

```diff
@@ -504,6 +508,28 @@ pub fn escape_bytes<W>(wr: &mut W, bytes: &[u8]) -> Result<()>
             b'\n' => b"\\n",
             b'\r' => b"\\r",
             b'\t' => b"\\t",
+            b'\x00' ... b'\x1F' | b'\x7F' => {
+                if start < i {
+                    try!(wr.write_all(&bytes[start..i]));
+                }
+
+                try!(write!(wr,"\\u{:04X}", *byte));
+
+                start = i + 1;
+
+                continue;
+            },
+            b'\x80' ... b'\x9F' if last_byte_was_c2 => {
```

Comment:
I absolutely agree and was about to open an issue to question the soundness of exposing a method which promise to encode binary content even when it is not supported by the standard.

Removing this method and only implementing escape_str() will provide these benefits:
- Doesn't give the false hope that the library can magically escape binary string despite the standard not allowing it. (Trying to do it anyway will result in UB)
- Guarantying that the generated JSON will be valid and parseable by other valid JSON implementations
- Simplifying the implementation which would be able to use [.chars()](https://doc.rust-lang.org/std/primitive.str.html#method.chars) like said above

In general I feel it to be more Rusty to enforce a sensible interface sticking to what the standard specifies and then to guaranty valid JSON generation.

I'll post new patches using [.chars()](https://doc.rust-lang.org/std/primitive.str.html#method.chars).